crates/sui-framework/packages/sui-framework/sources/crypto/group_ops.move

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Generic Move and native functions for group operations.
+module sui::group_ops {
+
+    // TODO: add .spec.move file
+
+    use std::vector;
+
+    friend sui::bls12381;
+    friend sui::ristretto255;
+
+    // TODO: remove before merging
+    use std::debug;
+
+    const EInvalidInput: u64 = 0;
+    const EInvalidBufferLength: u64 = 1;
+
+    /////////////////////////////////////////////////////
+    ////// Generic functions for group operations. //////
+
+    // The caller provides a type identifer that should match the types of enum [Groups] in group_ops.rs.
+
+    // General wrapper for all group elements.
+    struct Element<phantom T> has store, copy, drop {
+        bytes: vector<u8>,
+    }
+
+    public fun bytes<G>(e: &Element<G>): &vector<u8> {
+        &e.bytes
+    }
+    
+    public fun equal<G>(e1: &Element<G>, e2: &Element<G>): bool {
+        // TODO: Remove before merging
+        debug::print(&e1.bytes);
+        debug::print(&e2.bytes);
+        e1.bytes == e2.bytes
+    }
+
+    // Fails if the bytes are not a valid group element and 'is_trusted' is false.
+    public(friend) fun from_bytes<G>(type: u8, bytes: &vector<u8>, is_trusted: bool): Element<G> {
```

Comment:
so is_trusted is usually set to false if invoked externally?

Reply:
exactly